### ISSUE TITLE
KIE Sandbox Sample fails to build

### ISSUE THREAD

**Issue by dekaido** [apache/incubator-kie-tools#2213]
Build failure after creating Sample Kogito Quarkus process via the KIE Sandbox.
See example here: https://github.com/dekaido/traveler-process

Steps to reproduce in README.md
...

#### Create project from Sandbox
1. Open [KIE Sandbox](sandbox.kie.org/)
2. Click [Try Sample](https://sandbox.kie.org/#/import?url=https://sandbox.kie.org/samples/Sample.bpmn) under "Workflow"
3. Click "Apply Accelerator" > "Quarkus..." > Accept
4. Click "Share" > (Download) "All files"

#### Build local project (JVM)
1. Compile to run with JVM
      ``` log
      mvn clean package
      ```
2. Error messages
     ``` log
     [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.15.3.Final:build (default) on project quarkus-accelerator: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
      [ERROR] 	[error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateModel threw an exception: org.kie.memorycompiler.KieMemoryCompilerException: [org/kie/kogito/test/TravelersMessageProducer_7.java (24:88) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModel.java (42:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModel.java (44:27) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModel.java (48:45) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelOutput.java (42:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelOutput.java (44:27) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelOutput.java (48:45) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelInput.java (34:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelInput.java (36:27) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelInput.java (40:45) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (24:53) : package org.kie.kogito.addon.quarkus.messaging.common does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (28:49) : cannot find symbol
      [ERROR]   symbol: class QuarkusMessageConsumer, org/kie/kogito/test/TravelersMessageConsumer_3.java (28:103) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (44:56) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (53:48) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (62:112) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (73:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (73:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (87:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (87:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (147:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (147:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (151:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (151:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (168:102) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (41:52) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (52:5) : method does not override or implement a method from a supertype]
      [ERROR] 	at org.kie.memorycompiler.KieMemoryCompiler.compileNoLoad(KieMemoryCompiler.java:137)
      [ERROR] 	at org.kie.memorycompiler.KieMemoryCompiler.compileNoLoad(KieMemoryCompiler.java:105)
      [ERROR] 	at org.drools.drl.quarkus.util.deployment.DroolsQuarkusResourceUtils.compileGeneratedSources(DroolsQuarkusResourceUtils.java:153)
      [ERROR] 	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.createGeneratedBeanBuildItemsFromJavaSources(KogitoAssetsProcessor.java:273)
      [ERROR] 	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateModel(KogitoAssetsProcessor.java:182)
      [ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
      [ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
      [ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
      [ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
      [ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
      [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
      [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
      [ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
      [ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
      [ERROR] -> [Help 1]
     ```

#### Build local project (Quarkus)
1. Compile to run with Quarkus
      ``` log
      mvn clean compile quarkus:dev
      ```
2. Error messages
     ``` log
           2024-03-26 16:24:09,845 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
            [error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateModel threw an exception: org.kie.memorycompiler.KieMemoryCompilerException: [org/kie/kogito/test/TravelersMessageProducer_7.java (24:88) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModel.java (42:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModel.java (44:27) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModel.java (48:45) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelOutput.java (42:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelOutput.java (44:27) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelOutput.java (48:45) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelInput.java (34:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelInput.java (36:27) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersModelInput.java (40:45) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (24:53) : package org.kie.kogito.addon.quarkus.messaging.common does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (28:49) : cannot find symbol
        symbol: class QuarkusMessageConsumer, org/kie/kogito/test/TravelersMessageConsumer_3.java (28:103) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (44:56) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (53:48) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (62:112) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (73:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (73:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (87:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (87:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (147:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (147:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (151:28) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (151:67) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersProcess.java (168:102) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (41:52) : package org.acme.travel does not exist, org/kie/kogito/test/TravelersMessageConsumer_3.java (52:5) : method does not override or implement a method from a supertype]
            at org.kie.memorycompiler.KieMemoryCompiler.compileNoLoad(KieMemoryCompiler.java:137)
            at org.kie.memorycompiler.KieMemoryCompiler.compileNoLoad(KieMemoryCompiler.java:105)
            at org.drools.drl.quarkus.util.deployment.DroolsQuarkusResourceUtils.compileGeneratedSources(DroolsQuarkusResourceUtils.java:153)
            at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.createGeneratedBeanBuildItemsFromJavaSources(KogitoAssetsProcessor.java:273)
            at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateModel(KogitoAssetsProcessor.java:182)
            at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
            at java.base/java.lang.reflect.Method.invoke(Method.java:580)
            at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
            at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
            at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
            at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
            at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
            at java.base/java.lang.Thread.run(Thread.java:1583)
            at org.jboss.threads.JBossThread.run(JBossThread.java:501)

     ```

**Comment by tiagobento** [comment]
Thanks for reporting this. We have a new Sample with https://github.com/apache/incubator-kie-tools/pull/2409. Our `daily-dev` builds were failing until today, but I think it should be normalized tomorrow so you can test the new Sample at https://sandbox.kie.org/dev

**Comment by tiagobento** [comment]
Is this solved? Can I close this?

**Comment by dekaido** [comment]
@tiagobento hey, thanks for the updates. Looks like the build is fixed with the changes. I'm now getting a runtime error, which might be a separate issue.

Both `java -jar quarkus-run.jar` and `mvn clean compile quarkus:dev` return an j2ee Creation Exception:

``` console
ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
Caused by: jakarta.enterprise.inject.CreationException: Null contextual instance was produced by a normal scoped synthetic bean: SYNTHETIC bean [class=org.infinispan.client.hotrod.RemoteCacheManager, id=3e5ozT9gyWKQ9VvCN1iNcOXJaqs]
	at org.infinispan.client.hotrod.RemoteCacheManager_3e5ozT9gyWKQ9VvCN1iNcOXJaqs_Synthetic_Bean.doCreate(Unknown Source)
	at org.infinispan.client.hotrod.RemoteCacheManager_3e5ozT9gyWKQ9VvCN1iNcOXJaqs_Synthetic_Bean.create(Unknown Source)
	at org.infinispan.client.hotrod.RemoteCacheManager_3e5ozT9gyWKQ9VvCN1iNcOXJaqs_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.c74(Unknown Source)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
	at org.infinispan.client.hotrod.RemoteCacheManager_3e5ozT9gyWKQ9VvCN1iNcOXJaqs_Synthetic_ClientProxy.arc$delegate(Unknown Source)
	at org.infinispan.client.hotrod.RemoteCacheManager_3e5ozT9gyWKQ9VvCN1iNcOXJaqs_Synthetic_ClientProxy.administration(Unknown Source)
	at org.kie.kogito.infinispan.CacheProcessInstances.<init>(CacheProcessInstances.java:56)
	at org.kie.kogito.infinispan.AbstractProcessInstancesFactory.createProcessInstances(AbstractProcessInstancesFactory.java:39)
	at org.kie.kogito.infinispan.AbstractProcessInstancesFactory.createProcessInstances(AbstractProcessInstancesFactory.java:25)
	at org.kie.kogito.persistence.quarkus.InfinispanProcessInstancesFactory_ClientProxy.createProcessInstances(Unknown Source)
	at org.kie.kogito.process.impl.AbstractProcess.configure(AbstractProcess.java:183)
	at org.kie.kogito.process.impl.AbstractProcess.activate(AbstractProcess.java:203)
	at org.kie.kogito.hr.HiringProcess.<init>(HiringProcess.java:20)
	at org.kie.kogito.hr.HiringProcess_Bean.doCreate(Unknown Source)
	at org.kie.kogito.hr.HiringProcess_Bean.create(Unknown Source)
	at org.kie.kogito.hr.HiringProcess_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.c14(Unknown Source)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
	at org.kie.kogito.hr.HiringProcess_ClientProxy.arc$delegate(Unknown Source)
	at org.kie.kogito.hr.HiringProcess_ClientProxy.id(Unknown Source)
	at org.kie.kogito.app.Processes.setup(Processes.java:32)
	at org.kie.kogito.app.Processes_Bean.doCreate(Unknown Source)
	at org.kie.kogito.app.Processes_Bean.create(Unknown Source)
	at org.kie.kogito.app.Processes_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.c17(Unknown Source)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
	at org.kie.kogito.app.Processes_ClientProxy.arc$delegate(Unknown Source)
	at org.kie.kogito.app.Processes_ClientProxy.processIds(Unknown Source)
	at org.kie.kogito.services.registry.ProcessDefinitionEventRegistry.register(ProcessDefinitionEventRegistry.java:69)
	at org.kie.kogito.quarkus.registry.ProcessDefinitionRegistration.onStartUp(ProcessDefinitionRegistration.java:51)
	at org.kie.kogito.quarkus.registry.ProcessDefinitionRegistration_Observer_onStartUp_lne3cFiO5I9wh5H9OTf8cQM_oRc.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:346)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:328)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:82)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:155)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:106)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	... 11 more

```

**Comment by tiagobento** [comment]
This seems fixed on 10.0.0. I followed the reproducer steps and didn't get an error on `mvn clean compile`.

For `mvn clean compile quarkus:dev` I needed Docker runner (used Rancher Desktop with `export DOCKER_HOST=unix://$HOME/.rd/docker.sock`) and it took a while to download some images. The first attempt failed but once the images were already downloaded my second attempt succeeded and I was able to create a process instance via Swagger UI.

Closing.
